.github/workflows/upload-test-stats.yml

```diff
@@ -58,6 +58,14 @@ jobs:
           python3 -m tools.stats.upload_test_stats --workflow-run-id "${WORKFLOW_RUN_ID}" --workflow-run-attempt "${WORKFLOW_RUN_ATTEMPT}" --head-branch "${HEAD_BRANCH}"
           python3 -m tools.stats.upload_sccache_stats --workflow-run-id "${WORKFLOW_RUN_ID}" --workflow-run-attempt "${WORKFLOW_RUN_ATTEMPT}"
 
+      - name: Rerun if failed on master
+        if: github.event.workflow_run.run_attempt == 1 && github.event.workflow_run.head_branch == 'master'
+        env:
+          WORKFLOW_RUN_ID: ${{ github.event.workflow_run.id }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh run rerun "${WORKFLOW_RUN_ID}" --failed
```

Comment:
I wonder if this might interact with our metrics in some unexpected way because we only include `run_attempt == 1` in the calculation.  Specifically, retrying and the job successes, but then the `run_attempt` should be increased accordingly, so there is probably nothing to worry about.  Please just double check that